core/templates/dev/head/components/promo/PromoBarDirective.js

```diff
@@ -19,25 +19,36 @@
  */
 
 oppia.directive('promoBar', [
-  'UrlInterpolationService', function(UrlInterpolationService) {
+  'ResourceService', 'UrlInterpolationService', 'ENABLE_PROMO_BAR',
+  function(ResourceService, UrlInterpolationService, ENABLE_PROMO_BAR) {
     return {
       restrict: 'E',
-      scope: {
-        getPromoMessage: '&promoMessage'
-      },
       templateUrl: UrlInterpolationService.getDirectiveTemplateUrl(
         '/components/promo/' +
         'promo_bar_directive.html'),
       controller: [
-        '$scope',
-        function($scope) {
+        '$scope', '$window',
+        function($scope, $window) {
           var isPromoDismissed = function() {
             return !!angular.fromJson(sessionStorage.promoIsDismissed);
           };
           var setPromoDismissed = function(promoIsDismissed) {
             sessionStorage.promoIsDismissed = angular.toJson(promoIsDismissed);
           };
 
+          if ($window.location.pathname.split('/')[1] === 'signup') {
+            $scope.promoBarIsEnabled = false;
+            $scope.getPromoMessage = '';
```

Comment:
This naming seems wrong. Variables should be nouns; this name starts with a verb. You probably just want $scope.promoBarMessage.

Names that start with a verb should only be used to name functions/methods.

Reply:
Done.